cmd/cosign/cli/attach/sbom.go

```diff
@@ -60,6 +72,67 @@ func SBOMCmd(ctx context.Context, regOpts options.RegistryOptions, sbomRef strin
 	return remote.Write(dstRef, img, regOpts.GetRegistryClientOpts(ctx)...)
 }
 
+func sbomCmdOCIExperimental(ctx context.Context, regOpts options.RegistryOptions, sbomRef string, sbomType ocitypes.MediaType, imageRef string) error {
```

Comment:
This is a bit long for the `cmd/` package. Can we move some of it into `pkg/`?

There's also a fair bit of overlap with `SBOMCmd`. I bet if we reorder things you can actually stick the `if EnableOCIExperimental()` at the *end* of `SBOMCmd` and call this something like `writeSBOM`.